tap_wikipedia/utils/pipe.py

```diff
@@ -0,0 +1,21 @@
+from collections.abc import Callable, Iterable
+from functools import reduce
+
+from tap_wikipedia.models import wikipedia
+
+
+def pipe(
+    *,
+    callables: tuple[
+        Callable[[Iterable[wikipedia.Record]], Iterable[wikipedia.Record]], ...
+    ],
+    initializer: Iterable[wikipedia.Record],
+) -> Iterable[wikipedia.Record]:
+    """
+    Yield from a pipe of callables.
+
+    The callables are piped in linear order, from left-to-right.
+
+    The input to the first callable of the pipe is `initializer`.
+    """
+    yield from reduce(lambda x, y: y(x), callables, initializer)
```

Comment:
Nice-looking function, but there's probably something similar in itertools?
 https://docs.python.org/3/library/itertools.html#itertools.chain

Reply:
I don't see how `chain` can be used. The callables have to be applied cummulatively to transform `records`.

Reply:
Also, I deleted this file and moved the function to `WikipediaAbstractsStream.get_records`. The code looks cleaner to me this way.